collector/transport_node_collector.go

```diff
@@ -96,18 +112,27 @@ func (c *transportNodeCollector) generateTransportNodeMetrics() (transportNodeMe
 		}
 
 		var transportNodeType string
-		if e, ok := edgeClusterMap[transportNode.Id]; ok {
-			edgeClusterMembershipMetric := prometheus.MustNewConstMetric(c.edgeClusterMembership, prometheus.GaugeValue, 1.0, transportNode.Id, e.edgeClusterID, e.edgeMemberIndex)
-			transportNodeMetrics = append(transportNodeMetrics, edgeClusterMembershipMetric)
-			transportNodeType = "edge"
-		} else {
+		if edgeClusterMemberships != nil {
```

Comment:
related to line 75, we can check with `len(edgeClusterMemberships) > 0` instead

Reply:
But `len(edgeClusterMemberships)` can be 0 which not caused by error, when there's no edge cluster member. Is it possible for all edge cluster doesn't have a member? Or initially, there's no edge cluster?

Reply:
> But len(edgeClusterMemberships) can be 0
oh in this case then this is correct